### PR TITLE
Fixing REPL

### PR DESCRIPTION
**Pull request by luizperes**
I have fixed/implemented some of the aspects that were missing in our REPL, such as saving context
and breaking statements in multiple lines.

There is still a lot of work to do on it, but we can definitely use it for now. 

repl_test

### CODE REVIEW

File: src/ast/stmt/ProgramStmt.php [hunk 1]
```diff
@@ -59,4 +59,18 @@ public function runTypeChecker()
...
     }
+
+    public function attachValidAST($ast)
+    {
+        $safe_stmt_list = $this->stmt_list; // copy array
+        try {
+            $this->stmt_list = array_merge($this->stmt_list, $ast->stmt_list);
+            $this->injectScope($this->scope->parent);
+            $this->runTypeChecker();
+        } catch (\Exception $e) {
```
Comment (haskellcamargo) [inline src/ast/stmt/ProgramStmt.php:71]: To avoid namespace resolution, put a `use \Exception;` declaration.

File: src/ast/stmt/ProgramStmt.php [hunk 2]
```diff
@@ -59,4 +59,18 @@ public function runTypeChecker()
...
+            // rollback in case of error
+            $this->stmt_list = $safe_stmt_list;
+            echo $e;
```
Comment (haskellcamargo) [inline src/ast/stmt/ProgramStmt.php]: Is the `echo` operation meant to be here? Just for curiosity.

File: src/repl/QuackRepl.php [hunk 3]
```diff
@@ -101,24 +127,50 @@ function readline_callback($command)
             goto next;
     }
 
-    $lexer = new Tokenizer($command);
+    $run_command = !$session->complete_stmt
```
Comment (haskellcamargo) [inline src/repl/QuackRepl.php]: Prefer not using negations on conditions. The pattern `cond ? pass : fail` is easier to read than `!cond ? fail : pass`. :smile:

**Review by haskellcamargo — APPROVED** [review]

**Comment by luizperes** [comment]
I have just done those pointed out as well as solved one of the `TODO` comments for the REPL

File: src/ast/stmt/ProgramStmt.php (hunk 1 above)
Comment (haskellcamargo) [reply src/ast/stmt/ProgramStmt.php:71]: Now you can remove the `\` :smile:

**Review by haskellcamargo — APPROVED** [review]

**Pull request merged**